api/service/stagedstreamsync/syncing.go

```diff
@@ -156,67 +166,119 @@ func getBlockDbPath(beacon bool, loopID int, dbDir string) string {
 		}
 	} else {
 		if loopID >= 0 {
-			return fmt.Sprintf("%s_%d", filepath.Join(dbDir, "cache/blocks_db"), loopID)
+			return fmt.Sprintf("%s_%d_%d", filepath.Join(dbDir, "cache/blocks_db"), shardID, loopID)
 		} else {
-			return filepath.Join(dbDir, "cache/blocks_db_main")
+			return fmt.Sprintf("%s_%d", filepath.Join(dbDir, "cache/blocks_db_main"), shardID)
 		}
 	}
 }
 
+func (s *StagedStreamSync) Debug(source string, msg interface{}) {
+	// only log the msg in debug mode
+	if !s.config.DebugMode {
+		return
+	}
+	pc, _, _, _ := runtime.Caller(1)
+	caller := runtime.FuncForPC(pc).Name()
+	callerParts := strings.Split(caller, "/")
+	if len(callerParts) > 0 {
+		caller = callerParts[len(callerParts)-1]
+	}
+	src := source
+	if src == "" {
+		src = "message"
+	}
+	// SSSD: STAGED STREAM SYNC DEBUG
+	if msg == nil {
+		fmt.Printf("[SSSD:%s] %s: nil or no error\n", caller, src)
+	} else if err, ok := msg.(error); ok {
+		fmt.Printf("[SSSD:%s] %s: %s\n", caller, src, err.Error())
+	} else if str, ok := msg.(string); ok {
+		fmt.Printf("[SSSD:%s] %s: %s\n", caller, src, str)
+	} else {
+		fmt.Printf("[SSSD:%s] %s: %v\n", caller, src, msg)
+	}
+}
+
 // doSync does the long range sync.
 // One LongRangeSync consists of several iterations.
 // For each iteration, estimate the current block number, then fetch block & insert to blockchain
-func (s *StagedStreamSync) doSync(downloaderContext context.Context, initSync bool) (int, error) {
+func (s *StagedStreamSync) doSync(downloaderContext context.Context, initSync bool) (uint64, int, error) {
 
 	var totalInserted int
 
 	s.initSync = initSync
+	s.Debug("initSync", initSync)
 
 	if err := s.checkPrerequisites(); err != nil {
-		return 0, err
+		s.Debug("checkPrerequisites/error", err)
+		return 0, 0, err
 	}
 
 	var estimatedHeight uint64
 	if initSync {
 		if h, err := s.estimateCurrentNumber(downloaderContext); err != nil {
-			return 0, err
+			return 0, 0, err
 		} else {
 			estimatedHeight = h
 			//TODO: use directly currentCycle var
 			s.status.setTargetBN(estimatedHeight)
+			s.Debug("estimatedHeight", estimatedHeight)
 		}
 		if curBN := s.bc.CurrentBlock().NumberU64(); estimatedHeight <= curBN {
 			s.logger.Info().Uint64("current number", curBN).Uint64("target number", estimatedHeight).
 				Msg(WrapStagedSyncMsg("early return of long range sync"))
-			return 0, nil
+			return estimatedHeight, 0, nil
 		}
-
-		s.startSyncing()
-		defer s.finishSyncing()
 	}
 
+	s.startSyncing()
+	defer s.finishSyncing()
+
 	for {
 		ctx, cancel := context.WithCancel(downloaderContext)
 
 		n, err := s.doSyncCycle(ctx, initSync)
+		s.Debug("doSyncCycle/n", n)
```

Comment:
better log description? or let's remove if no longer required ?

Reply:
I am going to remove it.